TTS/tts/models/vits.py

```diff
@@ -1318,22 +1370,38 @@ def format_batch_on_device(self, batch):
         """Compute spectrograms on the device."""
         ac = self.config.audio
 
+        if self.args.TTS_part_sample_rate:
+            wav = self.audio_resampler(batch["waveform"])
+        else:
+            wav = batch["waveform"]
+
         # compute spectrograms
-        batch["spec"] = wav_to_spec(batch["waveform"], ac.fft_size, ac.hop_length, ac.win_length, center=False)
+        batch["spec"] = wav_to_spec(wav, ac.fft_size, ac.hop_length, ac.win_length, center=False)
```

Comment:
Why do we compute spec two times ?

Reply:
Because it is needed.  In the scenario of upsampling from 22khz to 44khz. `batch["spec"]` is computed using a 22khz audio and it will be used for the posterior encoder. On the other hand, the `spec_mel `will be computed using 44khz audio and will be used for the loss (that is computed using a 44khz ground truth audio). So, if `self.args.TTS_part_sample_rate is not None` we need to recompute the spectrogram using 44khz audio, otherwise, we can use the `batch["spec"] ` that is already in the same sampling rate.